File: src/2025-02-27-deepclaude37-rust/java/exercises/practice/resistor-color-trio/src/main/java/ResistorColorTrio.java
class ResistorColorTrio {
    String label(String[] colors) {
        int firstDigit = colorValue(colors[0]);
        int secondDigit = colorValue(colors[1]);
        int exponent = colorValue(colors[2]);

        long multiplier = calculateMultiplier(exponent);
        long value = (firstDigit * 10L + secondDigit) * multiplier;

        if (value >= 1_000_000_000 && value % 1_000_000_000 == 0) {
            return String.format("%d gigaohms", value / 1_000_000_000);
        } else if (value >= 1_000_000 && value % 1_000_000 == 0) {
            return String.format("%d megaohms", value / 1_000_000);
        } else if (value >= 1_000 && value % 1_000 == 0) {
            return String.format("%d kiloohms", value / 1_000);
        } else {
            return String.format("%d ohms", value);
        }
    }

    private static int colorValue(String color) {
        switch (color) {
            case "black": return 0;
            case "brown": return 1;
            case "red": return 2;
            case "orange": return 3;
            case "yellow": return 4;
            case "green": return 5;
            case "blue": return 6;
            case "violet": return 7;
            case "grey": return 8;
            case "white": return 9;
            default: throw new IllegalArgumentException("Invalid color: " + color);
        }
    }

    private static long calculateMultiplier(int exponent) {
        return (long) Math.pow(10, exponent);
    }
}

File: src/2025-02-27-deepclaude37-rust/javascript/exercises/practice/resistor-color-trio/resistor-color-trio.js
//
// This is only a SKELETON file for the 'Resistor Color Trio' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class ResistorColorTrio {
  static COLOR_VALUES = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white'];

  constructor(colors) {
    this.colors = colors;
  }

  get label() {
    const firstValue = ResistorColorTrio.COLOR_VALUES.indexOf(this.colors[0]);
    const secondValue = ResistorColorTrio.COLOR_VALUES.indexOf(this.colors[1]);
    const exponent = ResistorColorTrio.COLOR_VALUES.indexOf(this.colors[2]);
    
    if (firstValue === -1 || secondValue === -1 || exponent === -1) {
      throw new Error('invalid color');
    }
    
    let value = (firstValue * 10 + secondValue) * Math.pow(10, exponent);
    
    const prefixes = [
      { multiplier: 1e9, name: 'giga' },
      { multiplier: 1e6, name: 'mega' },
      { multiplier: 1e3, name: 'kilo' },
    ];
    
    let suffix = 'ohms';
    for (const { multiplier, name } of prefixes) {
      if (value >= multiplier && value % multiplier === 0) {
        value = value / multiplier;
        suffix = `${name}${suffix}`;
        break;
      }
    }
    
    return `Resistor value: ${value} ${suffix}`;
  }
}
